src/payment/strategies/stripe-upe/stripe-upe.ts

```diff
@@ -0,0 +1,236 @@
+     */
+    status: 'succeeded' | string;
+
+     /**
+      * The payment error encountered in the previous PaymentIntent confirmation. It will be cleared if the PaymentIntent is later updated for any reason.
+      */
+     last_payment_error: LastPaymentError | null;
+}
+
+/**
+ * The payment error encountered in the previous PaymentIntent confirmation. It will be cleared if the PaymentIntent is later updated for any reason.
+ */
+export interface LastPaymentError {
+    /**
+     * A human-readable message providing more details about the error. For card errors, these messages can be shown to your users.
+     */
+     message?: string;
+}
+
+export interface StripeError {
+    /**
+     * type of error. When the error type is card_error or validation_error, you can display the error message in error.message directly to your user.
+     */
+    type: string;
+    /**
+     * A human-readable message providing more details about the error. For card errors, these messages can be shown to your users.
+     */
+    message?: string;
+
+    /**
+     * The PaymentIntent object.
+     */
+    payment_intent: PaymentIntent;
+}
+
+/**
+ * Parameters that will be passed on to the Stripe API to confirm the PaymentIntent.
+ */
+export interface StripeUPEConfirmParams {
+    /**
+     * If you are [handling next actions yourself](https://stripe.com/docs/payments/payment-intents/verifying-status#next-actions), pass in a return_url. If the subsequent action
+     * is redirect_to_url, this URL will be used on the return path for the redirect.
+     *
+     * @recommended
+     */
+    return_url?: string;
+}
+
+/**
+ * Data to be sent with a `stripe.confirmPayment` request.
+ * Refer to the [Payment Intents API](https://stripe.com/docs/js/payment_intents/confirm_payment) for a full list of parameters.
+ */
+export interface StripeConfirmPaymentData {
+    /**
+     * The Elements instance that was used to create the Payment Element.
+     */
+    elements: StripeElements;
+
+    /**
+     * Parameters that will be passed on to the Stripe API to confirm the PaymentIntent.
+     */
+     confirmParams?: StripeUPEConfirmParams;
+
+    /**
+     * By default, confirmPayment will always redirect to your return_url after a successful confirmation.
+     * If you set redirect: "if_required", then confirmPayment will only redirect if your user chooses a redirect-based payment method.
+     */
+    redirect?: 'always' | 'if_required';
+}
+
+export interface StripeElement {
+    /**
+     * The `element.mount` method attaches your element to the DOM.
+     */
+    mount(domElement: string | HTMLElement): void;
+
+    /**
+     * Removes the element from the DOM and destroys it.
+     * A destroyed element can not be re-activated or re-mounted to the DOM.
+     */
+    destroy(): void;
+
+    /**
+     * Unmounts the element from the DOM.
+     * Call `element.mount` to re-attach it to the DOM.
+     */
+    unmount(): void;
+}
+
+export interface StripeElements {
```

Comment:
The same here

Reply:
I'm keeping this interface because it is different from the one stripev3 uses